rir/src/ir/insns.h

```diff
@@ -54,17 +54,29 @@ DEF_INSTR(ldfun_, 1, 0, 1, 0)
  */
 DEF_INSTR(ldvar_, 1, 0, 1, 0)
 
+/**
+ * ldvar_:: like ldvar.
+ * Stores an additional immediate with a unique number for the cache bindings.
+ */
+DEF_INSTR(ldvar_cache_, 2, 0, 1, 0)
```

Comment:
I would have prefered a lot, if we would not double the instructions. instead cache idx of 0 (or -1) could mean no caching.

Reply:
Ok, but do you have a strong reason? Because unless we have some "size" like argument it is usually better to have more instructions with particular semantics than adding an if to every load/store

Reply:
less code duplication (they all do almost the same. I mean in pir you chose to use the same instruction) and most will have a cache

Reply:
Ok, but the distinction was on purpose. PIR is not "executable" -> I do not mind executing ifs at compile time why I may mind execute an if !=-1 for every load/store RIR BC.